src/mailstoreobserver.cpp

```diff
@@ -466,8 +466,9 @@ void MailStoreObserver::removeMessages(const QMailMessageIdList &ids)
             _publicationChanges = true;
         }
     }
-    // Local action not handled by action observer
-    publishChanges();
+    if (_publicationChanges) {
```

Comment:
I'm not sure if this check is actually needed as publishChanges() checks it too, and without it here the behavior would match mailstoreChanges() name a bit better. No big deal, though.